signer/signer.go

```diff
@@ -26,6 +26,24 @@ type Subject struct {
 	Hosts []string   `json:"hosts"`
 }
 
+// SignRequest stores a signature request, which contains the hostname,
+// the CSR, optional subject information, and the signature profile.
+type SignRequest struct {
+	Hostname string   `json:"hostname"`
+	Request  string   `json:"certificate_request"`
+	Subject  *Subject `json:"subject,omitempty"`
+	Profile  string   `json:"profile"`
+	Label    string   `json:"label"`
+}
+
+// Root is used to define where the Signer gets its public certificate
+// and private keys for signing.
+type Root struct {
```

Comment:
Would you prefer *Path? If we want to include non-file backed keys, they can be added as new fields.